rackcity/views/datacenter_views.py

```diff
@@ -0,0 +1,32 @@
+from django.http import HttpResponse, JsonResponse
+from rackcity.models import Datacenter
+from django.core.exceptions import ObjectDoesNotExist
+from rackcity.api.serializers import DatacenterSerializer
+from rest_framework.decorators import permission_classes, api_view
+from rest_framework.permissions import IsAuthenticated, IsAdminUser
+from rest_framework.parsers import JSONParser
+from rest_framework.pagination import PageNumberPagination
+from http import HTTPStatus
+import math
+from io import StringIO
+
+
+@api_view(['GET'])
+@permission_classes([IsAuthenticated])
+def datacenter_all(request):
+    """
+        Return List of all datacenters.
+    """
+    try:
+        datacenters = Datacenter.objects.all()
+        serializer = DatacenterSerializer(datacenters, many=True)
+        return JsonResponse(
+            {"datacenters": serializer.data},
+            status=HTTPStatus.OK
+        )
+    except Datacenter.Error:
```

Comment:
Wasn't sure what error to catch so I just threw a standard error for now

Reply:
I wouldn't do this because it can lead to covering up an error with an incorrect failure message which can be super misleading. I'd say either
- figure out what the error type is when there are no datacenters (might be ObjectDoesNotExist or something like that) and catch that specifically
- don't catch it at all and just let the server 500. We don't want that but if it's an error we didn't know we could get then it should fail
- catch any error by saying `except Exception as error` and then make the failure message `str(error)`

Reply:
Also I think best practices is to have only the line you think might fail in the try: and then do the rest either in an else: or outside of the try/except block